src/description.jl

```diff
@@ -32,6 +32,10 @@ function description(dt::AbstractDateTime, p::Period, suffix::String)
         ts = string(max_val)
     end
 
+    if VERSION < v"0.7.0-DEV.3216"
```

Comment:
Typically when inside of a function, `VERSION` checks should be wrapped in `@static`.

Reply:
Thanks, I was unaware of that!

Reply:
Basically, when you use `@static`, the given condition is partially evaluated when the expression is parsed, and only the code in the conditional path taken is compiled when the function is compiled. If `@static` isn't used, the entire branch expression is compiled and the version check will be performed on every call (assuming the compiler doesn't optimize it out).